### ISSUE TITLE
Unit Conversions

### ISSUE THREAD

**Issue by Yoshod** [Yoshod/Sersi#167]
**Description**:
A cog which connects to an API for currency conversion, and mathematically carries out conversions where the rate is fixed. 

**Use Case**:
When users post messages saying something costs £50 that might not mean much to others, making the discussion less accessible. Or it could be someone stating it's really warm at 28c. This is the use case for this cog.

**Proposed Solution**:
We would create a message context command that searches the message for numbers proceeding or followed by a currency symbol or descriptor e.g. 50CAD. It would then convert this to EUR by default, unless the user has previously run a slash command changing their default currency. This would be stored in the database.

**Impact on Project**:
Would create a dependency on a currency conversion API library if a library is used.

**Priority**:
Major Feature

**Reason for Priority**:
This could take a long time to implement and test, and would require changes be made to the database.

**Milestone**:
Unsure

**Comment by gombik42** [comment]
should make it a cog for general unit conversions

**Comment by Yoshod** [comment]
> should make it a cog for general unit conversions

Agreed, however this would elevate from Minor to Major feature as that's a lot more to add.

**Comment by gombik42** [comment]
> > should make it a cog for general unit conversions
> 
> Agreed, however this would elevate from Minor to Major feature as that's a lot more to add.

I don't think that makes it necessarily a major feature, as you can just implement some conversions in the initial version and then add more late

**Comment by Yoshod** [comment]
Closing this as 5.4.0 is no longer planned.
